client/verta/verta/endpoint/_endpoint.py

```diff
@@ -295,15 +379,63 @@ def _get_or_create_stage(self, name="production"):
 
     def _create_stage(self, name="production"):
         url = "{}://{}/api/v1/deployment/workspace/{}/endpoints/{}/stages".format(
-            self._conn.scheme,
-            self._conn.socket,
-            self.workspace,
-            self.id
+            self._conn.scheme, self._conn.socket, self.workspace, self.id
         )
         response = _utils.make_request("POST", url, self._conn, json={"name": name})
         _utils.raise_for_http_error(response)
         return response.json()["id"]
 
+    def _set_kafka_settings(self, stage_id, kafka_settings):
+        if not isinstance(stage_id, six.integer_types):
+            raise TypeError("`stage_id` must be int, not {}".format(type(stage_id)))
+        if not isinstance(kafka_settings, KafkaSettings):
+            raise TypeError(
+                "`kafka_settings` must be of type verta.endpoint.KafkaSettings,"
+                " not {}".format(type(kafka_settings))
+            )
+
+        url = "{}://{}/api/v1/deployment/stages/{}/kafka".format(
+            self._conn.scheme,
+            self._conn.socket,
+            stage_id,
+        )
+        response = _utils.make_request(
+            "PUT", url, self._conn, json=kafka_settings._as_dict()
+        )
+        _utils.raise_for_http_error(response)
+
+    def _get_kafka_settings(self, stage_id):
+        if not isinstance(stage_id, six.integer_types):
+            raise TypeError("`stage_id` must be int, not {}".format(type(stage_id)))
+
+        url = "{}://{}/api/v1/deployment/stages/{}/kafka".format(
+            self._conn.scheme,
+            self._conn.socket,
+            stage_id,
+        )
+        response = _utils.make_request("GET", url, self._conn)
+        _utils.raise_for_http_error(response)
+
+        kafka_settings_dict = response.json().get("update_request", {})
+        if not kafka_settings_dict or kafka_settings_dict == {"disabled": True}:
+            return None
+        return KafkaSettings._from_dict(kafka_settings_dict)
+
+    def _del_kafka_settings(self, stage_id):
+        if not isinstance(stage_id, six.integer_types):
+            raise TypeError("`stage_id` must be int, not {}".format(type(stage_id)))
+
+        url = "{}://{}/api/v1/deployment/stages/{}/kafka".format(
+            self._conn.scheme,
+            self._conn.socket,
+            stage_id,
+        )
+        response = _utils.make_request(
+            "PUT", url, self._conn, json={"disabled": True},
+        )
+        _utils.raise_for_http_error(response)
+
+    # TODO: add support for KafkaSettings to _update_from_dict below or de-scope
```

Comment:
@daniel-verta Was this required as a part of the ticket?

Reply:
It was not. I realized that there was this pathway in the API for updating endpoints and left this as a note to explicitly not support updating the Kafka settings with this method, or to add such support. I think we should add a ticket tracking this, leave the reference here, and move on for the time being.